### PR TITLE
Fixed models for new data format

### PR DESCRIPTION
**Pull request by vovcyan**

### CODE REVIEW

File: src/main/java/com/ssu/schedule/parser/ParserSSU.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+            }
+
+            lessonsPerDay.addAll(parseLessons(obj.getString("id"), lessonArray));
+        }
+
+        return lessonsPerDay;
+    }
+
+    private ArrayList<Lesson> parseLessons(String id, JSONArray jsonLessons) throws JSONException {
+        ArrayList<Lesson> lessons = new ArrayList<>();
```
Comment (SevakAvet) [inline src/main/java/com/ssu/schedule/parser/ParserSSU.java:154]: List<Lesson> lessons = ...

File: src/main/java/com/ssu/schedule/parser/ParserSSU.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+            }
+
+            day.setWeekday(Integer.parseInt(id));
+            lesson.setDate(day);
+
+            switch (jsonLesson.getString("type")) {
+                case "practice":
+                    lesson.setType("Практика");
+                    break;
+            }
```
Comment (SevakAvet) [inline src/main/java/com/ssu/schedule/parser/ParserSSU.java:183]: switch with one case?

File: src/main/java/com/ssu/schedule/parser/ParserSSU.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    @Autowired
+    public ParserSSU(FacultyRepository facultyRepository, GroupRepository groupRepository) {
+        this.facultyRepository = facultyRepository;
+        this.groupRepository = groupRepository;
+    }
+
+    @Scheduled(cron = "0 0 12 * * ?")
+    private void getCurrentSchedule() {
+        OkHttpClient client = new OkHttpClient();
```
Comment (SevakAvet) [inline src/main/java/com/ssu/schedule/parser/ParserSSU.java:59]: you don't need to create client and requestBuilder for each request, create two beans instead and inject (@autowire) them whenever they're used

File: src/main/java/com/ssu/schedule/web/rest/ScheduleController.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        return getLastSchedule();
+    }
+
+    @Scheduled(cron = "0 20 12 1/1 * ?")
+    public void postScheduleToService() {
+        try {
+
+            OkHttpClient client = new OkHttpClient();
+
+            Gson gson = new GsonBuilder().create();
```
Comment (SevakAvet) [inline src/main/java/com/ssu/schedule/web/rest/ScheduleController.java:62]: The same as for okhttpclient (see comment above), create bean

**Review by SevakAvet — CHANGES_REQUESTED** [review]

File: src/main/java/com/ssu/schedule/web/rest/ScheduleController.java (hunk 4 above)
Comment (vovcyan) [reply src/main/java/com/ssu/schedule/web/rest/ScheduleController.java:62]: Ok

File: src/main/java/com/ssu/schedule/parser/ParserSSU.java (hunk 3 above)
Comment (vovcyan) [reply src/main/java/com/ssu/schedule/parser/ParserSSU.java:59]: Ok

File: src/main/java/com/ssu/schedule/parser/ParserSSU.java (hunk 1 above)
Comment (vovcyan) [reply src/main/java/com/ssu/schedule/parser/ParserSSU.java:154]: Ok

File: src/main/java/com/ssu/schedule/parser/ParserSSU.java (hunk 2 above)
Comment (vovcyan) [reply src/main/java/com/ssu/schedule/parser/ParserSSU.java:183]: Ok

**Pull request merged**